### PR TITLE
Add Muddle 54 and Zork

### LINKED ISSUE #1948: MUDSAV; SAV FILE and MUDSAV; FIXUP FILE
If you load a random Muddle image, and the interpreter exits with

```
:$ FATAL ERROR MAJOR FILE NOT FOUND $
```

... then it's failing to open `MUDSAV; SAV FILE` (or `MUDSAV; FIXUP FILE`). Your image is trying to call a compiled Muddle function from the system's library of pure functions - for example, Zork uses some functions from the `LSRTNS` library to get the user's real name.

There's a rather vague description of `SAV FILE` - the "special large file" - on p165 of the [Muddle 54 primer](https://raw.githubusercontent.com/PDP-10/muddle/master/doc/MDL_Primer_and_Manual.pdf). The code to search the library is in `MUDSYS; MAPPUR`. The code to initialise it and put functions into it is, inevitably, nowhere to be found...

An `RSUBR` object in Muddle can contain code in various forms. If it's a `PCODE`, that's a pointer into this library. The second word of a `PCODE` object is `offset-in-PURVEC,,offset-in-chunk`. When a pure `RSUBR` is called using Muddle's `MCALL` UUO, the UUO handler looks the code up in `PURVEC` and calls `PLOAD` to load it if it's not already mapped into memory.

`PURVEC` is a vector of library code chunks that this program might use, each being described by four words. `offset-in-PURVEC` above is an offset into this vector (so it's always a multiple of 4).
- Word 0: the SIXBIT name of the chunk.
- Word 1: where it's mapped in the current process, or 0 if it isn't.
- Word 2: (something to do with the LRU expiry process for getting rid of unused mappings)
- Word 3: where it can be found in `SAV FILE` - ``length-in-pages,,first-page`` - or 0 if we haven't looked yet.

To load a chunk from `SAV FILE`, you do the following:
- Take the last (non-zero) character in the chunk's name, divide it by the first word of `SAV FILE`, take the remainder and add one.
- That gives you a word number in the first page of `SAV FILE`. Read that word and it gives you a "directory" page number.
- Read the directory page. The first word of that page is the length of the directory in words. The rest is a sorted vector of pairs of words, where the first word is the chunk name you're looking for, and the second contains:
  - top bit - unused?
  - 7 bits - length of the chunk in pages
  - 10 bits - Muddle version number
  - right half - number of the first page in `SAV FILE`
- Binary-search the directory to find the chunk you want, map the pages into memory, and update the `PURVEC` entry.

If a chunk with a matching name and version number isn't found in `SAV FILE`, Muddle will then look for `MUDTMP; (name) SAV(version)` (e.g. `MUDTMP; LSRTNS SAV54`). If *that* isn't found, it'll look at `FIXUP FILE`, which contains patches for pure code that was originally compiled for a different version of Muddle. It has the same structure as `SAV FILE`, but the data is fixup information rather than code. So the final fallback is to load a chunk for a different version from `SAV FILE`, and apply a patch from `FIXUP FILE`.

In order to be able to run Zork images without patching them, we would need to generate a `SAV FILE` that contains a stub version of `LSRTNS` (that just returns `FALSE` from `LSR-ENTRY`), and (I think) a valid but empty `FIXUP FILE`. This shouldn't be particularly difficult...

### PR DESCRIPTION
**Pull request by atsampson**
The overall aim here is to make it possible to run the 1977/8 Zork save files that MIT recently released (#1811) on ITS. These commits:

- Add reconstructed source for the Muddle 54 interpreter, based on the various source versions we've got, and the executables that @larsbrinkhoff and @rmaldersoniii found (#1940). in particular we didn't previously have the GC file for ITS; we do now.
- Fix some problems in Muddle 56 revealed by the above.
- Add a program to generate MUDSAV; SAV FILE and FIXUP FILE, with stubs for the functions Zork uses.
- Add the Zork documentation (from Panda), the latest ITS Zork save file we have, and TAA's Zork launcher.

Fixes #1948. Still doesn't achieve #832 but some of this will help!

### CODE REVIEW

**Review by eswenson1 — APPROVED** [review]

**Comment by larsbrinkhoff** [comment]
Hats off to @atsampson!  I'm so glad Zork is running now.

MKSVFL is written in the DM style, nice touch.

File: doc/madman/madadv.help [hunk 1]
```diff
@@ -0,0 +1,331 @@
+ZORK User Information
+
+I. Getting a ZORK
+ 
+    a. ITS:  Say ZORK^K to DDT rather than :ZORK to get a zork.
```
Comment (larsbrinkhoff) [inline doc/madman/madadv.help:5]: Any clue what the difference between ^K and a colon command would be?  When I looked in DDT, I didn't see a difference.

**Comment by larsbrinkhoff** [comment]
Commit message "Both of these are from MIT's zork-1978-12 release", should be "zork-1978-01".

**Review by larsbrinkhoff — APPROVED** [review]

**Comment by larsbrinkhoff** [comment]
I fixed the "zork-1978-12" commit message.

**Comment by larsbrinkhoff** [comment]
I'm monitoring the CI builds.

**Comment by larsbrinkhoff** [comment]
@atsampson, would you like to write something to its-hackers?

File: doc/madman/madadv.help (hunk 1 above)
Comment (atsampson) [reply doc/madman/madadv.help:5]: If you already have a ZORK job, ZORK^K will get rid of it before starting a new one. I guess they were worried about people accidentally running multiple copies.

**Comment by atsampson** [comment]
Done.

**Pull request merged**